### PR TITLE
VB-3723 Add age config to VSIP prison API and database

### PR DESCRIPTION
**Pull request by AledWEvans**
Added new config for prisons to allow for 
Max visitors and adult age

### CODE REVIEW

**Review by dhirajshettymoj — COMMENTED** [review]
comments / queries added

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/UpdatePrisonDto.kt [hunk 1]
```diff
@@ -7,8 +7,20 @@ import jakarta.validation.constraints.Min
 data class UpdatePrisonDto(
   @Schema(description = "minimum number of days notice from the current date to booked a visit", example = "2", required = false)
   @field:Min(0)
-  var policyNoticeDaysMin: Int?,
+  val policyNoticeDaysMin: Int?,
```
[the remark this answers is not in the document]
Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/UpdatePrisonDto.kt:10]: Narr we don't do this, it's the approach we have used with the other attributes

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/PrisonDto.kt [hunk 2]
```diff
@@ -16,10 +16,23 @@ data class PrisonDto(
...
   @field:Min(0)
-  var policyNoticeDaysMin: Int,
+  val policyNoticeDaysMin: Int,
   @Schema(description = "maximum number of days notice from the current date to booked a visit", example = "28", required = true)
   @field:Min(0)
-  var policyNoticeDaysMax: Int,
+  val policyNoticeDaysMax: Int,
+
+  @Schema(description = "Max number of total visitors")
+  @field:Min(1)
```
[the remark this answers is not in the document]
Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/PrisonDto.kt:31]: added to all elements on prison dto,...

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/PrisonDto.kt [hunk 3]
```diff
@@ -16,10 +16,23 @@ data class PrisonDto(
...
+  val maxTotalVisitors: Int,
+  @Schema(description = "Max number of adults")
+  @field:Min(1)
+  val maxAdultVisitors: Int,
+  @Schema(description = "Max number of children, if -1 then no limit is applied")
+  @field:Min(-1)
+  val maxChildVisitors: Int,
+  @Schema(description = "Age of adults in years")
+  @field:Min(10)
```
[the remark this answers is not in the document]
Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/PrisonDto.kt:43]: Well 16,... might be seen as adults

[the remark this answers is not in the document]
Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/PrisonDto.kt:43]: not sure with this one TBO

[the remark this answers is not in the document]
Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/PrisonDto.kt:43]: I will ask @Simon

**Review by dhirajshettymoj — APPROVED** [review]
LGTM - will bring up the minimum age for adults tomorrow in standup

**Pull request merged**